src/pywifes/splice.py

```diff
@@ -0,0 +1,330 @@
+import astropy.io.fits as fits
+import numpy as np
+import scipy.sparse as sp
+
+
+def is_evenly_sampled(_x):
+    dx = _x[1:] - _x[:-1]
+    return abs((dx.max() - dx.min()) / np.median(dx)) < 1e-3
+
+
+def _A_lanczos(_x, x, a=2, missing=np.nan):
+    """
+    Returns transformation matrix for Lanczos (sinc) interpolation
+
+    Inputs
+       _x:  x-values at which the original function is sampled
+        x:  x-values to which we want to interpolate
+        a:  integer order of the kernel
+        missing:  value to use for missing data (default:  np.nan)
+    Outputs
+        A:  sparse representation of the transformation matrix
+    """
+    if not is_evenly_sampled(_x):
+        print("_A_lanczos:  warning -- _x not evenly sampled!")
+        print("Lanczos interpolation may not give good results.")
+    # Figure out which x values are in range
+    x = np.atleast_1d(x)
+    ok = np.all([x >= _x[0], x < _x[-1]], axis=0)
+    # Find bins and fractional positions within bins, and interpolate
+    fi = np.array((len(_x) - 1) * (x - _x[0]) / (_x[-1] - _x[0]))
+    i = np.array([int(f) for f in fi])
+    # Write down the matrix for Lanczos interpolation
+    EPSILON = np.finfo(np.double).eps
+    # The following line of code results in a deprecation waring
+    # A sparce matrix in link list format
+    A = sp.lil_matrix((len(x), len(_x)))
+    for m in np.arange(len(x)):
+        # If x[m] is in interpolation range...
+        if ok[m]:
+            # Generate a range of i within the original signal
+            ik = range(max(0, i[m] - a + 1), min(len(_x) - 1, i[m] + a + 1))
+            # Find L(x-i) for each of these
+            xk = np.pi * (fi[m] - np.array(ik) + EPSILON)
+            w = a * np.sin(xk) * np.sin(xk / a) / xk**2
+            A[m, ik] = w / w.sum()
+        elif x[m] == _x[-1]:
+            A[m, -1] = 1
+        else:
+            A[m, 0] = missing
+    return A
+
+
+# Unclear if needed
+class SingleSpec(object):
```

Comment:
A class like this could be a nice fix for the complexity we're running into in `join_cubes`.

Maybe something similar, storing most of the same data as here, though maybe going further and adding functions to compute and store things like AB, flux_B, diag_B, given a particular row and column?? 

Might be easier than trying to rework the functions the other ways I've suggested.